examples/rabbit_fox_hawk.jl

```diff
@@ -0,0 +1,410 @@
+norm(vec) = √sum(vec .^ 2)
+
+# The environment is generated from a heightmap: a 2D matrix, where each value denotes the
+# height of the terrain at that point. We segregate the model into 4 regions based on the
+# height:
+# - Anything below `water_level` is waster and cannot be walked on
+# - The region between `water_level` and `grass_level` is flatland, that can be walked on
+# - The part of the map between `grass_level` and `mountain_level` is too high for animals to
+#   walk over, but it can be flown over
+# - The terrain above `mountain_level` is completely inaccessible
+#
+# The `initialize_model` function takes in the URL to our heightmap, the thresholds for the 4
+# regions, and some additional parameters for the model. It then creates and returns a model
+# with the specified heightmap and containing the specified number of rabbits, foxes and hawks.
+
+function initialize_model(
+    heightmap_url,
+    water_level = 10,
+    grass_level = 20,
+    mountain_level = 35;
+    n_rabbits = 120,  ## initial number of rabbits
+    n_foxes = 60,  ## initial number of foxes
+    n_hawks = 60,  ## initial number of hawks
+    Δe_grass = 4,  ## energy gained from eating grass
+    Δe_rabbit = 40,  ## energy gained from eating one rabbit
+    rabbit_repr = 0.03,  ## probability for a rabbit to (asexually) reproduce at any step
+    fox_repr = 0.04,  ## probability for a fox to (asexually) reproduce at any step
+    hawk_repr = 0.04, ## probability for a hawk to (asexually) reproduce at any step
+    rabbit_vision = 3,  ## how far rabbits can see grass and spot predators
+    fox_vision = 9,  ## how far foxes can see rabbits to hunt
+    hawk_vision = 20,  ## how far hawks can see rabbits to hunt
+    regrowth_chance = 0.01,  ## probability that a patch of grass regrows at any step
+    seed = 42,  ## seed for random number generator
+)
+
+    ## Download and load the heightmap. The grayscale value is converted to `Float64` and
+    ## scaled from 1 to 40
+    heightmap = floor.(Int, convert.(Float64, load(download(heightmap_url))) * 39) .+ 1
+    ## The dimensions of the space is that of the heightmap
+    dims = (size(heightmap)..., 40)
+    ## Generate the RNG for the model
+    rng = MersenneTwister(seed)
+
+    space = GridSpace(dims; periodic = false)
+
+    ## The region of the map that is accessible to each type of animal (land-based or flying)
+    ## is defined using `BitArrays`
+    land_walkmap = BitArray(falses(dims...))
+    air_walkmap = BitArray(falses(dims...))
+    for i in 1:dims[1], j in 1:dims[2]
+        ## land animals can only walk on top of the terrain between water_level and grass_level
+        water_level < heightmap[i, j] < grass_level &&
+            (land_walkmap[i, j, heightmap[i, j]+1] = true)
+        ## air animals can fly at any height upto mountain_level
+        heightmap[i, j] < mountain_level &&
+            (air_walkmap[i, j, (heightmap[i, j]+1):mountain_level] .= true)
+    end
```

Comment:
This is more readable with standard `if` clauses. My suggestion is that if you need more than 1 lines, use a proper codeblock instead of going through the one-liner syntaxes like `&&` or `? :`